test/suites/drift/suite_test.go

```diff
@@ -955,4 +955,112 @@ var _ = Describe("Drift", func() {
 			env.ConsistentlyExpectNoDisruptions(int(numPods), time.Minute)
 		})
 	})
+	Context("Capacity Reservations", func() {
+		var largeCapacityReservationID, xlargeCapacityReservationID string
+		BeforeAll(func() {
+			largeCapacityReservationID = aws.ExpectCapacityReservationCreated(
+				env.Context,
+				env.EC2API,
+				ec2types.InstanceTypeM5Large,
+				env.ZoneInfo[0].Zone,
+				1,
+				nil,
+				nil,
+			)
+			xlargeCapacityReservationID = aws.ExpectCapacityReservationCreated(
+				env.Context,
+				env.EC2API,
+				ec2types.InstanceTypeM5Xlarge,
+				env.ZoneInfo[0].Zone,
+				1,
+				nil,
+				nil,
+			)
+		})
+		AfterAll(func() {
+			aws.ExpectCapacityReservationsCanceled(env.Context, env.EC2API, largeCapacityReservationID, xlargeCapacityReservationID)
+		})
+		BeforeEach(func() {
+			nodePool.Spec.Template.Spec.Requirements = []karpv1.NodeSelectorRequirementWithMinValues{{
+				NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+					Key:      karpv1.CapacityTypeLabelKey,
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{karpv1.CapacityTypeReserved},
+				},
+			}}
+		})
+		It("should drift nodeclaim when the reservation is no longer selected by the nodeclass", func() {
+			nodeClass.Spec.CapacityReservationSelectorTerms = []v1.CapacityReservationSelectorTerm{{ID: largeCapacityReservationID}}
+			pod := coretest.Pod()
+			env.ExpectCreated(nodePool, nodeClass, pod)
+			nc := env.EventuallyExpectNodeClaimCount("==", 1)[0]
+			env.EventuallyExpectNodeClaimsReady(nc)
+			n := env.EventuallyExpectCreatedNodeCount("==", 1)[0]
+			Expect(n.Labels).To(HaveKeyWithValue(corev1.LabelInstanceTypeStable, string(ec2types.InstanceTypeM5Large)))
+			Expect(n.Labels).To(HaveKeyWithValue(karpv1.CapacityTypeLabelKey, karpv1.CapacityTypeReserved))
+			Expect(n.Labels).To(HaveKeyWithValue(v1.LabelCapacityReservationID, largeCapacityReservationID))
+
+			nodeClass.Spec.CapacityReservationSelectorTerms = []v1.CapacityReservationSelectorTerm{{ID: xlargeCapacityReservationID}}
```

Comment:
Does this create any flakes/races with Node/NodeClaim registration because we might not cleanup this test properly if something is actively drifting? I think we may need the same do-not-disrupt pod here

Reply:
Oh, maybe. You're thinking if a node comes up after we do the check in the AfterEach? Adding the do-not-disrupt annotation wouldn't hurt.